docs/Solidity.g4

```diff
@@ -371,21 +371,27 @@ subAssembly
   : 'assembly' identifier assemblyBlock ;
 
 numberLiteral
-  : (DecimalNumber | HexNumber) NumberUnit? ;
+  : (DecimalNumberWithoutExponent | DecimalNumberWithExponent | HexNumber) NumberUnit? ;
 
 identifier
   : ('from' | 'calldata' | 'address' | Identifier) ;
 
 BooleanLiteral
   : 'true' | 'false' ;
 
-DecimalNumber
-  : ( DecimalDigits | (DecimalDigits? '.' DecimalDigits) ) ( [eE] '-'? DecimalDigits )? ;
+DecimalNumberWithoutExponent
+  : ( DecimalDigits | (DecimalDigits? '.' DecimalDigits) ) ;
+
+DecimalNumberWithExponent
+  : ( DecimalDigits | (DecimalDigits? '.' DecimalDigits) ) ( [eE] '-'? DecimalDigits );
 
 fragment
 DecimalDigits
   : [0-9] ( '_'? [0-9] )* ;
 
+VersionLiteral
+  : [0-9]+ '.' [0-9]+ '.' [0-9]+ ;
```

Comment:
What about `0.5` - is that parsed as a fractional number? I'm not sure I like that...

Reply:
Yes. It is ugly. I'm trying to comfort myself thinking that a fractional number can denote a version the same way as an integer number.